sklearn/utils/_array_api.py

```diff
@@ -320,6 +331,8 @@ def _remove_non_arrays(*arrays, remove_none=True, remove_types=(str,)):
             continue
         if sp.issparse(array):
             continue
+        if _is_pandas_df_or_series(array):
+            continue
```

Comment:
We probably need something similar for any type of dataframe/series: pandas, polars, pyarrow.